src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -685,5 +687,35 @@ private static bool IsFrameworkIncompatible(Framework sourceFramework, Framework
             }
             return !sourceFramework.Name.Equals(targetFramework.Name, StringComparison.OrdinalIgnoreCase);
         }
+
+        /// <summary>
+        /// Returns true if source platform is incompatible with target platform.
+        /// </summary>
+        private static bool IsPlatformArchitectureIncompatible(Architecture sourcePlatform, Architecture targetPlatform)
+        {
+            if (sourcePlatform == Architecture.X86 && targetPlatform == Architecture.X64 ||
```

Comment:
There are different platforms(X86, X64, ARM, AnyCPU)
If we simply check for not equals, any of the incompatibility will be marked as true. 
This might cause unnecessary incompatibility. For example `source = Architecture.X64` is okay to run with `target = Architecture.AnyCPU`.
Do we want to change that behavior?